src/pages/Home/Homepage.js

```diff
@@ -1,30 +1,56 @@
 import { Carousel } from 'components/logicComponents/Carousel';
 import './styles.scss';
 import useTranslation from 'hooks/useTranslation';
+import Button from 'components/common/Button';
+import { useEffect, useState } from 'react';
 
 import { mockArrayData } from 'mock/mockData';
 
 export function HomePage() {
   const t = useTranslation();
+  const [buttonValue, setButtonValue] = useState('discover');
+  const [buttonState, setButtonState] = useState({
+    trending: false,
+    discover: false,
+    top_rated: false,
+  });
+
+  const filterMovies = async e => {
+    setButtonValue(e.target.name);
+  };
 
   return (
     <div className="Homepage-container">
       <div className="Homepage-container__header"></div>{' '}
-      <div className="Homepage-container__carousel">
-        <div className="Homepage-container__carousel-container">
-          {' '}
-          <Carousel movies={mockArrayData} />{' '}
-        </div>
-        <div className="Homepage-container__Watchlist-title">
-          {' '}
-          <h1> {t('homepage.titles.watchlist')} </h1>
-          <h3> {t('homepage.titles.list')} </h3>
+      <div className="Homepage-container__content">
+        <div className="Homepage-container__carousel">
+          <div className="Homepage-container__button">
+            {' '}
+            <Button name="trending" handleClick={e => filterMovies(e)}>
```

Comment:
What about this? So we don't need the arrow function
```suggestion
            <Button name="trending" handleClick={filterMovies}>
```

Reply:
Correcting that i realize that i had a bug where the state didn't changed as planned, turning into "undefined" not showing any carousel items for some reason, changing some things i realize that this is the better way to fix the problem:      
 <Button handleClick={() => filterMovies('trending')}> ===>  passing by parameter the actual value, it seems that the event.target.value sometimes comes with some delay